src/lib/components/tablev2/useSyncedScroll.ts

```diff
@@ -0,0 +1,38 @@
+import React, { useEffect, useRef } from "react";
+import { Row } from "react-table";
+import { FixedSizeList } from "react-window";
+
+export default function useSyncedScroll<
+  DATA_ROW extends Record<string, unknown> = Record<string, unknown>,
+>(): { headerRef: React.MutableRefObject<HTMLDivElement>, bodyRef: React.MutableRefObject<FixedSizeList<Row<DATA_ROW>[]>>}{
+  const headerRef = useRef<HTMLDivElement | null>(null);
+  const bodyRef = useRef<FixedSizeList<Row<DATA_ROW>[]> | null>(null);
+  const currentHeadRef = headerRef.current;
+  const currentBodyRef = bodyRef.current;
+  useEffect(() => {
+    if (bodyRef.current && headerRef.current) {
+      const listener = (event: Event) => {
+        headerRef.current.scrollTo({
+          left: (event.target as HTMLDivElement).scrollLeft,
+          top: 0,
+        });
+      };
+
+      /*
+      We intentionally use _outerRef prop here despite the fact that it is 
+      internal use only and not typed, as it is the only way for us to access to the scrollable element
+      */
+      //@ts-expect-error
+      (bodyRef.current._outerRef as HTMLDivElement).addEventListener(
+        'scroll',
+        listener,
+      );
+      return () => {
+        //@ts-expect-error
+        bodyRef.current._outerRef.removeEventListener('scroll', listener);
+      };
+    }
+  }, [currentHeadRef, currentBodyRef]);
```

Comment:
@ognyan-kostadinov I would suggest to try the below approach to solve the issue we were discussing today:
```suggestion
  const [listener, setListener] = useState<(event: Event) => void | null>(null);
  const [tableBody, setTableBody] = useState<FixedSizeList<Row<DATA_ROW>[]> | null>(null);
  const headerRef = (element: HTMLDivElement) => {
      const listener = (event: Event) => {
        element.scrollTo({
          left: (event.target as HTMLDivElement).scrollLeft,
          top: 0,
        });
      };
      setListener(listener);
  }
  const bodyRef = (tableBody: FixedSizeList<Row<DATA_ROW>[]> ) => {
    setTableBody(tableBody);
  };
  
  useEffect(() => {
    if(tableBody && listener) {
    /*
      We intentionally use _outerRef prop here despite the fact that it is 
      internal use only and not typed, as it is the only way for us to access to the scrollable element
      */
      //@ts-expect-error
      (tableBody._outerRef as HTMLDivElement).addEventListener(
        'scroll',
        listener,
      );
      return () => {
        //@ts-expect-error
        bodyRef.current._outerRef.removeEventListener('scroll', listener);
      };
      }
  }, [tableBody, listener])
  ```